server/aws-lsp-codewhisperer/src/shared/supplementalContextUtil/unitTestIntentDetection.ts

```diff
@@ -0,0 +1,146 @@
+    java: [
+        /@Test/m,
+        /import\s+org\.junit\./m,
+        /import\s+org\.testng\./m,
+        /import\s+org\.mockito\./m,
+        /^import\s+org\.assertj\./m,
+        /^import\s+org\.springframework\.test\./m,
+    ],
+    typescript: [/describe\(/m, /(it|test)\(/m],
+    javascript: [/describe\(/m, /(it|test)\(/m],
+}
+
+export class TestIntentDetector {
+    constructor() {}
+
+    detectUnitTestIntent(doc: TextDocument): boolean {
+        const lang = doc.languageId
+        const isTestFile = this.isTestFile(doc.uri, doc.getText(), lang)
+        return isTestFile
+    }
+
+    // @VisibleForTesting
+    isTestFile(filePath: string, fileContent: string, language: string): boolean {
+        if (!testFileNameRegex[language]) {
+            throw new Error('lang not supported by utg completion')
+        }
+
+        // Filename + file extension
+        const basename = path.basename(filePath)
+        const isTestFileByName = testFileNameRegex[language].some(regex => regex.test(basename))
+        // Return early and no need to inspect further
+        if (!isTestFileByName) {
+            return false
+        }
+
+        return testKeywordsRegex[language].some(regex => regex.test(fileContent))
+    }
+
+    // The following methods are not used for now as we found they're too strict for test intent detection, could remove once we confirm we no longer need them
+    // @VisibleForTesting
+    javaTestIntent(content: string): boolean {
+        const signaturePattern = new RegExp(
+            '@Test(?:\\s*@\\w+(?:\\(.*?\\))?\\s*)*' +
+                '(?:\\s*(?:public|protected|private)\\s+)?' +
+                '(?:static\\s+)?' +
+                '\\s*void\\s+\\w+\\s*\\([^)]*\\)' +
+                '(?:\\s*throws\\s+\\w+(?:,\\s*\\w+)*)?' +
+                '\\s*\\{',
+            'gm'
+        )
+
+        return this.curlyBracesSyntaxUtil(signaturePattern, content)
+    }
+
+    // @VisibleForTesting
+    jsTsTestIntent(content: string): boolean {
+        const signaturePattern = new RegExp(
+            `(it|test)\\s*\\(\\s*["\\'].*?["\\']\\s*,\\s*(async\\s+)?\\(\\s*\\)\\s*=>\\s*\\{`,
+            'gm'
+        )
+
+        return this.curlyBracesSyntaxUtil(signaturePattern, content)
+    }
+
+    // @VisibleForTesting
+    pyTestIntent(content: string): boolean {
+        const pattern = new RegExp(
+            'def\\s+test_\\w+\\s*\\(.*\\):',
+            'gms' // g: global, m: multiline, s: dotall
+        )
+
+        // Find all matches
+        const matches = [...content.matchAll(pattern)]
+
+        if (matches.length === 0) {
+            return false
+        }
+
+        // Get content after the last test
+        const lastMatch = matches[matches.length - 1]
+        const lastMatchPos = lastMatch.index! + lastMatch[0].length
+        const tailFromLastTest = content.slice(lastMatchPos)
+        const lines = tailFromLastTest.split('\n')
```

Comment:
should use either `\r?\n` or `os.EOL`

Reply:
nm these are not used at this point. Will consider removing them when we're sure they are 100% not needed anymore